### DISCUSSION TITLE
member.is_on_mobile should be shifted to discord.Status

### DISCUSSION THREAD

**Discussion by Hunter2809** [Rapptz/discord.py#7315] [Ideas]
I was going through docs, and found out about the `member.is_on_mobile()` method. But, it can be incorrect sometimes, as it depends on the `status` of the `member`. So, in my opinion, it should be under `discord.Status` class instead of `discord.Member` as it is a status of of the `member`

What are your thoughts on this?

**Comment by ghost** [comment]
`discord.Status` is merely an enum representing a member's status. It knows nothing about the member.

**Reply by Hunter2809** [reply]
Yes, I know about that. But, `is_on_mobile()` just checks if the user has a mobile status, and it can be wrong sometimes. Suppose, when I am on mobile but have my status as DND. I don't really think it fits in the `discord.Member` class as it checks for the status of the member and `discord.Status` is a thing for that...

**Reply by scrazzz** [reply]
> So, in my opinion, it should be under discord.Status class 

As nickofolas said, `discord.Status` is an enum, not a class.

> But, is_on_mobile() just checks if the user has a mobile status, and it can be wrong sometimes.

How can "it be wrong sometimes" ?

> Suppose, when I am on mobile but have my status as DND.

Yes, we wouldn't know if the you're on mobile or not if your status is set to DND on mobile. Hence, `Member.is_on_mobile` returns `True` even if the member's status is set to DND (or any other statuses) and is on mobile.
